storage/dune

```diff
@@ -1,76 +1,74 @@
 (library
-  (name xcp_storage_interface_types)
-  (public_name xapi-idl.storage.interface.types)
-  (flags (:standard -w -39))
-  (modules vdi_automaton)
-  (libraries
-    rpclib.core
-    threads
-    xapi-idl
-  )
-  (wrapped false)
-  (preprocess (pps ppx_sexp_conv ppx_deriving_rpc))
-)
+ (name xcp_storage_interface_types)
+ (public_name xapi-idl.storage.interface.types)
+ (flags (:standard -w -39))
+ (modules vdi_automaton)
+ (libraries
+   rpclib.core
+   threads
+   xapi-idl
+ )
+ (wrapped false)
+ (preprocess (pps ppx_sexp_conv ppx_deriving_rpc)))
 
 (library
-  (name xcp_storage_interface)
-  (public_name xapi-idl.storage.interface)
-  (flags (:standard -w -39))
-  (modules storage_interface)
-  (libraries
-    astring
-    rpclib.core
-    threads
-    xapi-stdext-date
-    xapi-idl
-    xapi-idl.storage.interface.types
-  )
-  (wrapped false)
-  (preprocess (pps ppx_sexp_conv ppx_deriving_rpc))
-)
+ (name xcp_storage_interface)
+ (public_name xapi-idl.storage.interface)
+ (flags (:standard -w -39))
+ (modules storage_interface)
+ (libraries
+   astring
+   rpclib.core
+   threads
+   xapi-stdext-date
+   xapi-idl
+   xapi-idl.storage.interface.types
+ )
+ (wrapped false)
+ (preprocess (pps ppx_sexp_conv ppx_deriving_rpc)))
 
 (library
-  (name xcp_storage)
-  (public_name xapi-idl.storage)
-  (flags (:standard -w -39))
-  (modules (:standard \ storage_interface storage_test vdi_automaton suite vdi_automaton_test))
-  (libraries
-    rpclib.core
-    threads
-    xapi-stdext-date
-    xapi-idl
-    xapi-idl.storage.interface
-  )
-  (wrapped false)
-  (preprocess (pps ppx_sexp_conv ppx_deriving_rpc))
-)
+ (name xcp_storage)
+ (public_name xapi-idl.storage)
+ (flags (:standard -w -39))
+ (modules
+  (:standard \ storage_interface storage_test vdi_automaton suite
+    vdi_automaton_test))
+ (libraries
+   rpclib.core
+   threads
+   xapi-idl
+   xapi-idl.storage.interface
+   xapi-stdext-date
+ )
+ (wrapped false)
+ (preprocess (pps ppx_sexp_conv ppx_deriving_rpc)))
 
 (executable
-  (name storage_test)
-  (flags (:standard -w -39))
-  (modules storage_test)
-  (libraries
-    cmdliner
-    alcotest
-    xapi-idl
-    xapi-idl.storage
-  )
-  (preprocess (pps ppx_sexp_conv ppx_deriving_rpc))
-)
+ (modes byte exe)
+ (name storage_test)
+ (flags (:standard -w -39))
+ (modules storage_test)
+ (libraries
+   cmdliner
+   alcotest
+   xapi-idl
+   xapi-idl.storage
+ )
+ (preprocess (pps ppx_sexp_conv ppx_deriving_rpc)))
 
 (executable
-  (name suite)
-  (flags (:standard -w -39))
-  (modules suite vdi_automaton_test)
-  (libraries
-    alcotest
-    xapi-idl.storage.interface
-    xapi-idl.storage.interface.types
-  )
-)
+ (modes byte exe)
```

Comment:
I think we should run the unit tests using the native executable since that is what we use in production (bytecode libraries are useful for `utop`, but we shouldn't be producing bytecode executables for things that we install, or things we run as part of unit tests).

If you just use `(modes exe)` I think it'll mean `best exe`, so it'll use native if available and fall back to bytecode only if your compiler doesn't support native.

Reply:
The default modes is `(modes exe)` I think removing this line is safe as long as we don't want to build the bytecode exe, which doesn't seem like something we want to do anyway.